.github/workflows/main.yml

```diff
@@ -155,6 +157,11 @@ jobs:
 
   publish:
     name: Publish Binaries
+    if: |
+      always()
```

Comment:
i'd like to just understand what and why you're doing this. maybe we can chat?

Reply:
https://github.com/openziti/ziti/issues/1087

Reply:
@dovholuknf  There's a bit more context and a detailed explanation of this workaround in this issue: https://github.com/openziti/ziti/issues/1087

Reply:
maybe a comment would be in order here, explaining the workaround? I'd expect most folks who look at this critically would at least be confused by the `always()`, and might even remove it.

Reply:
Comments added to reduce confusion for future maintainers, and I gave the reasons for skipping certain builds some more thought. 

#### TL;DR

I've refactored this CI workflow to always build Windows and macOS, even on forks, and allow forks to opt-out of those builds by setting a GH Actions repo var.

#### Explanation

I agree it makes sense to continue building Windows and macOS by default, as long as they can succeed, despite being time-consuming. 

I didn't specifically mention that I wanted to skip some builds because they were time-consuming when we talked because I was attempting to generalize a solution for ziti and zrok. The case I did mention where a signing cert is required is applicable to zrok's branch builds and ziti release builds, not ziti branch builds.

I've changed the conditional to allow forks to opt out of time-consuming builds by setting an optional var in their repo's GH Actions "vars." Ziti and zrok (different PR for zrok in the works) builds that require a signing cert will only run if repo owner is "openziti."

```
    if: vars.ZITI_SKIP_MACOS_BUILD != 'true'
```

With these vars set, the fork's CI run looks like this example in my fork where I'm now able to iterate quickly and obtain a container image with the Linux build: https://github.com/qrkourier/ziti/actions/runs/4766536885